distributed/utils_test.py

```diff
@@ -2419,3 +2420,110 @@ async def wait_for_state(
         # message as an exception wouldn't work.
         print(msg)
         raise
+
+
+_ports_in_use = set()
+
+
+@pytest.fixture()
+def name_of_test(request):
+    return f"{request.node.nodeid}"
+
+
+try:
+    # If we're not running with pytest-xdist we'll need to register this testrun_uid
+    import xdist  # noqa
+except ImportError:
+
+    @pytest.fixture(scope="session")
+    def testrun_uid():
+        return None
+
+
+@pytest.fixture(scope="session", autouse=True)
+def port_lock_filename(testrun_uid, tmpdir_factory):
+    if testrun_uid:
+        tmpdir = tmpdir_factory.mktemp("_distributed_test_port_lock", numbered=False)
+
+        yield tmpdir / testrun_uid
+    else:
+        yield None
+
+
+@pytest.fixture(scope="session")
+def global_port_lock(port_lock_filename):
+    if port_lock_filename:
+        import locket
+
+        lock = locket.lock_file(port_lock_filename)
+        yield lock
+    else:
+
+        @contextmanager
+        def dummy_lock():
+            yield
+
+        yield dummy_lock()
+
+
+@contextmanager
+def _get_open_port(global_port_lock):
+    while True:
+        with global_port_lock:
+            port = open_port()
+            if port not in _ports_in_use:
+                _ports_in_use.add(port)
+                break
+    try:
+        yield port
+    finally:
+        _ports_in_use.discard(port)
+
+
+@pytest.fixture()
+def free_port(global_port_lock, name_of_test):
+    with _get_open_port(global_port_lock) as port:
+        print(f"Using free port {port} for test {name_of_test}")
+        yield str(port)
```

Comment:
why do you need to change it to str?

Reply:
https://pypi.org/project/ephemeral-port-reserve/